src/pages/Oauth/Login.tsx

```diff
@@ -0,0 +1,174 @@
+import KyberOauth2, { LoginFlow, LoginMethod } from '@kybernetwork/oauth2'
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+import Loader from 'components/Loader'
+import { didUserReject } from 'constants/connectors/utils'
+import { ENV_KEY } from 'constants/env'
+import { useActiveWeb3React, useWeb3React } from 'hooks'
+import useParsedQueryString from 'hooks/useParsedQueryString'
+import { Container, Content, KyberLogo, TextDesc } from 'pages/Oauth/styled'
+import getShortenAddress from 'utils/getShortenAddress'
+import { queryStringToObject } from 'utils/string'
+import { formatSignature } from 'utils/transaction'
+
+import AuthForm from './AuthForm'
+import { createSignMessage, getSupportLoginMethods } from './helpers'
+
+const getErrorMsg = (error: any) => {
```

Comment:
consider to update `friendlyError` instead of create new parse error function

Reply:
I prefer to keep it.
 I want to see the origin error msg. easy for us to debug when users capture errors if they can not sign in.